### DISCUSSION TITLE
CppUTest::MemoryOperatorOverloadTest::deleteUnallocatedMemory generates warning -Wfree-nonheap-object

### DISCUSSION THREAD

**Discussion by j3270** [cpputest/cpputest#1777] [General]
The generated warning makes sense, what doesn't make sense to me is why even have the line 

delete (char*) 0x1234678;

when the following line is,

FAIL("Should never come here"); // LCOV_EXCL_LINE

Isn't that adequate for the test purposes?
Does the test really need to delete an arbitrary address? 

I'm raising the discussion because I don't like to see warnings in the build and that is the only one I am getting.

**Comment by thetic** [comment]
The intent of this test is to exercise some error handling in CppUTest's memory leak detection feature. However, as far as I can tell, `delete (char*) 0x1234678;` is undefined behavior regardless of the use of a custom `delete` implementation. This test is known to hang when compiled with clang-cl. I have #1676 open to remove this test. In the meantime, I recommend disabling CppUTest's leak detection for your usage.

**Comment by basvodde** [comment]
Ok ok, I merged the open PR.
